rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/ui/shaders/ShaderWidgetFactory.kt

```diff
@@ -0,0 +1,16 @@
+package com.jetbrains.rider.plugins.unity.ui.shaders
+
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.wm.StatusBar
+import com.intellij.openapi.wm.StatusBarWidget
+import com.intellij.openapi.wm.StatusBarWidgetFactory
+import com.jetbrains.rider.UnityProjectDiscoverer
+
+class ShaderWidgetFactory: StatusBarWidgetFactory {
+    override fun getId() = "ShaderWidget"
+    override fun isAvailable(project: Project) = UnityProjectDiscoverer.getInstance(project).isUnityProject
+    override fun canBeEnabledOn(statusBar: StatusBar) = true
+    override fun getDisplayName() = "Shader Context"
```

Comment:
This is shown whenever you right click on the status bar, so something like "Unity Shader File Context" would help when it's not visible, or in non-Unity projects.